lib/more_core_extensions/core_ext/array/tableize.rb

```diff
@@ -118,10 +121,42 @@ def format_row(row, widths, justifications)
       end
 
       def format_field(field, width, justification)
-        field = field.to_s.gsub(/\n|\r/, '').slice(0, width)
+        field = field.to_s.gsub(/\n|\r/, '')
+        field = ansi_truncate(field, width)
         "%0#{justification}#{width}s" % field
       end
 
+      def ansi_escapes?(field)
+        field.match?(ANSI_ESCAPE_SEQUENCE)
```

Comment:
What do you think of `!!(field =~ ANSI_ESCAPE_SEQUENCE)` to get the tests passing on older rubies?